icicle/appUtils/msm/msm.cu

```diff
@@ -907,26 +904,59 @@ namespace msm {
 
   } // namespace
 
+  MSMConfig DefaultMSMConfig()
+  {
+    device_context::DeviceContext ctx = {
+      0,               // device_id
+      (cudaStream_t)0, // stream
+      0,               // mempool
+    };
+    MSMConfig config = {
+      false, // are_scalars_on_device
+      false, // are_scalars_montgomery_form
+      0,     // points_size
+      1,     // precompute_factor
+      false, // are_points_on_device
+      false, // are_points_montgomery_form
+      1,     // batch_size
+      false, // are_results_on_device
+      0,     // c
+      0,     // bitsize
+      false, // is_big_triangle
+      10,    // large_bucket_factor
+      false, // is_async
+      ctx,   // DeviceContext
+    };
+    return config;
+  }
+
   template <typename S, typename A, typename P>
   cudaError_t MSM(S* scalars, A* points, int msm_size, MSMConfig config, P* results)
   {
+    int bitsize = (config.bitsize == 0) ? S::NBITS : config.bitsize;
     // TODO: DmytroTym/HadarIngonyama - unify the implementation of the bucket method and the batched bucket method in
     // one function
-    // TODO: DmytroTym/HadarIngonyama - parameters to be included into the implementation: on deviceness of points,
-    // scalars and results, precompute factor, points size and device id
     if (config.batch_size == 1)
       bucket_method_msm(
-        config.bitsize, config.c, scalars, points, msm_size, results, config.are_scalars_on_device, config.big_triangle,
-        config.large_bucket_factor, config.ctx.stream);
+        bitsize, 16, scalars, points, msm_size, results, config.are_scalars_on_device,
+        config.are_scalars_montgomery_form, config.are_points_on_device, config.are_points_montgomery_form,
+        config.are_results_on_device, config.is_big_triangle, config.large_bucket_factor, config.is_async,
+        config.ctx.stream);
     else
       batched_bucket_method_msm(
-        config.bitsize, config.c, scalars, points, config.batch_size, msm_size, results, config.are_scalars_on_device,
-        config.ctx.stream);
+        bitsize, (config.c == 0) ? get_optimal_c<S>(bitsize) : config.c, scalars, points, config.batch_size, msm_size,
+        results, config.are_scalars_on_device, config.ctx.stream);
     return cudaSuccess;
```

Comment:
I don't see us using `config->point_size` even though the comments on MSMConfig say it is used instead of msm_size in certain cases. This parameter in the config is a bit confusing, can't we always just use msm_size and batch_size to get the correct values?

Reply:
You're right, it's currently unused, I tried to mark this fact with the following comment: https://github.com/ingonyama-zk/icicle/pull/262/files#diff-cdd3b67200a359e20797b18b99ca7176f36d4c9d48d9843d22500c5424fe1c7dR106. The idea of this parameter is to control whether you want to re-use the same set of bases in all MSMs or have custom bases for each one. Current default is the latter, although it's really easy to make this parameter functional and from my understanding this will help you in halo2 where you could reduce memory footprint from bases by sharing bases between MSMs, right?